doc/src/modules/core.rst

```diff
@@ -46,9 +46,19 @@ singleton
 
 S
 ^
-.. autoclass:: S
+.. autoclass:: Singleton
    :members:
 
+.. note:: This example is worth reading.
+
+   * S.Zero is the same thing as Integer(0) the S object makes it 
```

Comment:
I think it's helpful to have such a comment.

I'm not sure how the "note" appears, but rather than say that it's worth reading you might say something to catch the reader's attention like the following:

```
.. note:: What's the difference between `x == 1` and `x is S.One`?

Most Integers -- even identical ones like Integer(2) and Integer(2) -- get stored at arbitrary memory
locations and tests for equality between them and a value must be done with the `==` operator: `x == 2`.
S,Zero and S.One are singleton objects, however, that return values of Integer(0) and Integer(1), respectively,
and are always stored at one memory location. Testing for equality with them can be done as `x == 1` but 
is more efficiently done as `x is S.One`. The reason that 0 and 1 are stored in this way is because of their
frequent use throughout the codebase.
```